src/Microsoft.AspNetCore.Razor.Runtime/Runtime/TagHelpers/TagHelperExecutionContext.cs

```diff
@@ -214,6 +214,53 @@ public void AddTagHelperAttribute(string name, object value)
             _allAttributes.Add(name, value);
         }
 
+        /// <summary>
+        /// Clears the <see cref="TagHelperExecutionContext"/> and updates its state with the provided values.
+        /// </summary>
+        /// <param name="tagName">The tag name to use.</param>
+        /// <param name="tagMode">The <see cref="TagMode"/> to use.</param>
+        /// <param name="items">The <see cref="IDictionary{Object, Object}"/> to use.</param>
+        /// <param name="uniqueId">The unique id to use.</param>
+        /// <param name="executeChildContentAsync">The <see cref="Func{Task}"/> to use.</param>
+        public void Reinitialize(
```

Comment:
Suggest removing overlapping parameters from the constructors, requiring a `Reinitialize()` call for all uses. Alternatively call `Reinitialize()` from the constructors duplicating this code.